src/codelabs/async-await/index.md

```diff
@@ -0,0 +1,452 @@
+  style="margin-top: 25px; margin-bottom: 25px"
+  frameborder="no"
+  height="350"
+  width="100%" >
+</iframe>
+
+In this example, `getUserOrder` completes with an Exception indicating that the user ID is invalid.
+
+You have learned about futures and how they complete, but how do you handle their
+results? In the next section you will learn how to use the `async` and `await`
+keywords to handle the results of asynchronous functions. 
+
+{{ site.alert.secondary }}
+  Quick Review:
+
+  * In Dart, a
+  [Future]({{site.dart_api}}/Future-class.html)
+  produces a result of type `<T>`. For example,
+  [Future]({{site.dart_api}}/Future-class.html)`<String>`
+  is the type signature for a future that produces a string result. If a
+  future produces a result that isn't a usable value, then the future's type is
+  `Future<void>`.
+  * A future can be in one of two states: uncompleted, or completed. 
+  * A completed future can either (1) complete with a result or (2) complete with
+  an error.
+  * When you invoke a function that returns a future, two things happen:
+
+  1. The function queues up work to be done and returns an uncompleted future.
+  2. Later, when the operation is finished, the future completes with a value or
+  with an error.
+{{ site.alert.end }}
+
+
+## Working with futures in Dart: async and await
+
+The `async` and `await` keywords provide a declarative way to define 
+asynchronous functions and access their results *after they have completed.*
+
+To define an `async` function:
+
+1. Add the `async` keyword after the function's parameters but before the function's body:  
+    {% prettify dart %}
+        void getUserOrder() [!async!] {
+    {% endprettify %}
+
+2. Update the function's type signature to return a future. The following  
+example defines an `async` function that doesn't return a usable value:
+    {% prettify dart %}
+        [!Future<void>!] getUserOrder() async {
+    {% endprettify %}
+
+{{ site.alert.info }}
+ You can 
+ [use Future\<void\>]({{url}}/guides/language/effective-dart/design#do-use-futurevoid-as-the-return-type-of-asynchronous-members-that-do-not-produce-values)
+ as the return type for asynchronous functions that don't return usable values.
+{{ site.alert.end }}
+
+Now that you have declared an async function:
+  1. Your function returns a future that completes with the type that you specify.
+  1. You must use the `await` keyword before invoking the function  to "wait"
+   for the future to complete either with a value or an error.
+
+{% prettify dart %}
+var order = [!await!] getUserOrder();
+{% endprettify %}
+
+
+The following example provides a side-by-side comparison of a synchronous function
+and an asynchronous function. Since you can only use the `await` keyword within 
+an `async` function body, the example converts the parent function from 
+synchronous to asynchronous thus enabling use of the `await` keyword: 
```

Comment:
And, since I was already confused by the first sentence, this second sentence made no sense. I would move this thought to be just in front of the 2nd DartPad and say something like:

The following example shows the same code as before, but implemented asynchronously. Since you can only use the `await` keyword within an `async` function body, the `createOrderMessage()` function is now tagged as `async`.